File: ios/Sources/MetaPixelPlugin/MetaPixelPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(MetaPixelPlugin)
public class MetaPixelPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "MetaPixelPlugin"
    public let jsName = "MetaPixel"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = MetaPixel()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}

File: android/src/main/java/com/columnapp/plugins/metapixel/MetaPixelPlugin.java
package com.columnapp.plugins.metapixel;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "MetaPixel")
public class MetaPixelPlugin extends Plugin {

    private MetaPixel implementation = new MetaPixel();

    @PluginMethod
    public void getAnonymousID(PluginCall call) throws NoSuchMethodException {
        JSObject ret = new JSObject();
        ret.put("value", implementation.getAnonymousID());
        call.resolve(ret);
    }
}
